src/D_configure_taskserver.sh

```diff
@@ -1,197 +1,198 @@
-	echo $output
+	TASKD_DATA=$1
+	output=$(taskd config --force server.key "$TASKD_DATA"/server.key.pem --data "$TASKD_DATA")
+	echo "$output"
 }
 
 configure_taskd_for_server_crl() {
-	TASKDDATA=$1
-	output=$(taskd config --force server.crl $TASKDDATA/server.crl.pem --data $TASKDDATA)
-	echo $output
+	TASKD_DATA=$1
+	output=$(taskd config --force server.crl "$TASKD_DATA"/server.crl.pem --data "$TASKD_DATA")
+	echo "$output"
 }
 
 configure_taskd_for_ca_cert() {
-	TASKDDATA=$1
-	output=$(taskd config --force ca.cert $TASKDDATA/ca.cert.pem --data $TASKDDATA)
-	echo $output
+	TASKD_DATA=$1
+	output=$(taskd config --force ca.cert "$TASKD_DATA"/ca.cert.pem --data "$TASKD_DATA")
+	echo "$output"
 }
 
-configure_taskserver_log() {
-	TASKDDATA=$1
+configure_task_server_log() {
+	TASKD_DATA=$1
 	PWD=$2
-	output=$(taskd config --force log $PWD/taskd.log)
-	echo $output
+	output=$(taskd config --force log "$PWD"/taskd.log)
+	echo "$output"
 }
 
-configure_taskserver_pid() {
-	TASKDDATA=$1
+configure_task_server_pid() {
+	TASKD_DATA=$1
 	PWD=$2
-	output=$(taskd config --force pid.file $PWD/taskd.pid)
-	echo $output
+	output=$(taskd config --force pid.file "$PWD"/taskd.pid)
+	echo "$output"
 }
 
-configure_taskserver_ip_and_port() {
+configure_task_server_ip_and_port() {
 	IP=$1
 	PORT=$2
-	output=$(taskd config --force server $IP:$PORT)
-	echo $output
+	output=$(taskd config --force server "$IP":"$PORT")
+	echo "$output"
 }
 
 show_taskd_config() {
 	output=$(taskd config)
-	echo $output
+	echo "$output"
 }
 
 
 start_taskdctl() {
 	output=$(taskdctl start)
-	echo $output
+	echo "$output"
 }
 
 show_running_taskdctl() {
-	output=$(ps -leaf | grep taskd)
-	echo $output
+	output=$(ps -leaf | pgrep taskd)
+	echo "$output"
 }
 
-function swapline_containing_string {
-  local OLD_LINE_PATTERN=$1; shift
-  local NEW_LINE=$1; shift
-  local FILE=$1
-  local NEW=$(echo "${NEW_LINE}" | sed 's/\//\\\//g')
+swapline_containing_string() {
+  _local OLD_LINE_PATTERN="$1"; shift
+  _local NEW_LINE="$1"; shift
+  _local FILE="$1"
+  _local NEW="$(echo "${NEW_LINE}" | sed 's/\//\\\//g')"
   touch "${FILE}"
-  sed -i '/'"${OLD_LINE_PATTERN}"'/{s/.*/'"${NEW}"'/;h};${x;/./{x;q100};x}' "${FILE}"
-  if [[ $? -ne 100 ]] && [[ ${NEW_LINE} != '' ]]
+  # shellcheck disable=SC2154
+  sed -i "/${OLD_LINE_PATTERN}/{s/.*/${NEW}/;h};${x;/./{x;q100};x}" "${FILE}"
+  if [ $? -ne 100 ] && [ "${NEW_LINE}" != '' ]
   then
     echo "${NEW_LINE}" >> "${FILE}"
   fi
 }
 
 modify_taskd_service_file(){
-	taskddata=$1
-	echo "taskddata=$taskddata" >&2
+	taskd_data=$1
+	echo "taskd_data=$taskd_data" >&2
 	
 	linux_username=$(whoami)
 	echo "linux_username=$linux_username" >&2
 	linux_group=$linux_username
 	echo "linux_group=$linux_group" >&2
 	
-	swapline_containing_string "ExecStart=" "ExecStart=/usr/local/bin/taskd server --data $taskddata" "src/taskd.service"
+	swapline_containing_string "ExecStart=" "ExecStart=/usr/_local/bin/taskd server --data $taskd_data" "src/taskd.service"
```

Comment:
I am wondering what the underscore does, you can ignore this comment, but if you would like to explain its function, I would greatly appreciate it!

Reply:
`In POSIX sh, 'local' is undefined.`
https://github.com/koalaman/shellcheck/wiki/SC2039